src/lib/tecs/TECS.hpp

```diff
@@ -49,351 +50,685 @@
+				      const Flag &flag) const;
 
-	void set_ste_rate_time_const(float time_const) { _STE_rate_time_const = time_const; }
-	void set_speed_derivative_time_constant(float time_const) { _speed_derivative_time_const = time_const; }
+	/**
+	 * @brief Update controlled throttle setpoint.
+	 *
+	 * @param dt is the update time intervall in [s].
+	 * @param se is the calculated specific energy.
+	 * @param flag is the control flags.
+	 */
+	void _calcThrottleControl(float dt, const SpecificEnergyRates &se, const Param &param, const Flag &flag);
 
-	void set_seb_rate_ff_gain(float ff_gain) { _SEB_rate_ff = ff_gain; }
+	/**
+	 * @brief Calculate throttle control specific total energy
+	 *
+	 * @param limit is the specific total energy rate limits.
+	 * @param se is the specific energy rates.
+	 * @param param is the control parameters.
+	 * @return EnergyRateValues
+	 */
+	EnergyRateValues _calcThrottleControlSte(const STELimit &limit, const SpecificEnergyRates &se,
+			const Param &param) const;
 
-	// TECS status
-	uint64_t timestamp() { return _pitch_update_timestamp; }
-	ECL_TECS_MODE tecs_mode() { return _tecs_mode; }
+	/**
+	 * @brief Calculate the throttle control update function.
+	 * Update the throttle control states.
+	 *
+	 * @param dt is the update time intervall in [s].
+	 * @param limit is the specific total energy rate limits.
+	 * @param ste is the specific total energy rates.
+	 * @param param is the control parameters.
+	 * @param flag is the control flags.
+	 */
+	void _calcThrottleControlUpdate(float dt, const STELimit &limit, const EnergyRateValues &ste, const Param &param,
+					const Flag &flag);
 
-	float hgt_setpoint() { return _hgt_setpoint; }
-	float vert_pos_state() { return _vert_pos_state; }
+	/**
+	 * @brief Calculate the throttle control output function.
+	 *
+	 * @param limit is the specific total energy rate limits.
+	 * @param ste is the specific total energy rates.
+	 * @param param is the control parameters.
+	 * @param flag is the control flags.
+	 * @return throttle setpoin in [0,1].
+	 */
+	float _calcThrottleControlOutput(const STELimit &limit, const EnergyRateValues &ste, const Param &param,
+					 const Flag &flag) const;
 
-	float TAS_setpoint_adj() { return _TAS_setpoint_adj; }
-	float tas_state() { return _tas_state; }
-	float getTASInnovation() { return _tas_innov; }
+private:
+	// State
+	AlphaFilter<float> _ste_rate_error_filter;		///< Low pass filter for the specific total energy rate.
+	float _pitch_integ_state{0.0f};				///< Pitch integrator state [rad].
+	float _throttle_integ_state{0.0f};			///< Throttle integrator state.
+
+
+	// Output
+	DebugOutput _debug_output;				///< Debug output.
+	float _pitch_setpoint{0.0f};				///< Controlled pitch setpoint [rad].
+	float _throttle_setpoint{0.0f};				///< Controlled throttle setpoint [0,1].
+	float _percent_undersped{0.0f};				///< A continuous representation of how "undersped" the TAS is [0,1]
```

Comment:
Calling it percent but form [0, 1] is maybe misleading. Could `_ratio_underspeed` work too?